utils/minio_utils.py

```
@@ -0,0 +1,163 @@
+            recursive (bool): Whether to list objects recursively in directories
+
+        Returns:
+            Generator yielding object names
+        """
+        try:
+            objects = self.client.list_objects(
+                self.bucket_name, prefix=prefix, recursive=recursive
+            )
+            return (obj.object_name for obj in objects)
+        except Exception as e:
+            raise Exception(f"Failed to list objects: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)